examples/react-whiteboard/src/App.js

```diff
@@ -45,7 +45,7 @@ function Canvas({ shapes }) {
 
   const insertRectangle = () => {
     batch(() => {
-      const shapeId = Date.now();
+      const shapeId = Date.now().toString();
```

Comment:
On v6 protocol, we enforce that a parentKey should be a string, but our whiteboard example/tuto actually use a number. This one is actually quite bad... I'm to tired to think of a solution but it's something we need to figure out before releasing 0.17. cc @ofoucherot @nvie

Reply:
Ah, good catch. Curious how you figured out what went wrong with this. I assume these messages would be silently dropped/ignored by the backend, causing weird behavior in the app? Was it immediately obvious to you what went wrong? Without TypeScript support, these may be hard to catch indeed!

The solution that would solve the issue at the core is to introduce a runtime type check in the place where we generate the `Op` instance, and possibly allow `LiveMap.set()` to take a number key as well, in which case we would simply convert it to a string before building the `Op` instance.

I can make that fix!

Reply:
Small update here: I went looking to fix this in the client, but quickly realized this a much deeper rabbit hole, and warrants a separate quality pass of its own.

This is not the only case that we need to address. We can tackle this particular issue as a one-off fix now, sure, but the _class_ fix for this would be to make sure that _all_ public APIs always perform runtime type checking on their arguments. (We wouldn't have to do this for internal APIs; we have TypeScript for that.)

This would be very maintenance unfriendly and error prone to be enforcing manually, so I'd suggest to enforce this using some kind of build step or (custom) rollup plugin.

For now, though, the fix for this #339.